homeassistant/components/generic/config_flow.py

```diff
@@ -173,26 +162,43 @@ async def async_test_still(
             response = await async_client.get(url, auth=auth, timeout=GET_IMAGE_TIMEOUT)
             response.raise_for_status()
             image = response.content
+            content_type = response.headers["content-type"]
     except (
         TimeoutError,
         RequestError,
         HTTPStatusError,
         TimeoutException,
+        KeyError,
     ) as err:
         _LOGGER.error("Error getting camera image from %s: %s", url, type(err).__name__)
         return {CONF_STILL_IMAGE_URL: "unable_still_load"}, None
 
     if not image:
         return {CONF_STILL_IMAGE_URL: "unable_still_load"}, None
-    fmt = get_image_type(image)
-    _LOGGER.debug(
-        "Still image at '%s' detected format: %s",
-        info[CONF_STILL_IMAGE_URL],
-        fmt,
-    )
-    if fmt not in SUPPORTED_IMAGE_TYPES:
+    if not content_type.startswith("image/"):
+        return {CONF_STILL_IMAGE_URL: "invalid_still_image"}, None
+
```

Comment:
Do we need this check? We run it through PIL anyway?

Reply:
Done